metrics/ci_integrations/lib/jenkins/adapter/jenkins_ci_client_adapter.dart

```diff
@@ -0,0 +1,182 @@
+import 'package:ci_integration/jenkins/client/jenkins_client.dart';
+import 'package:ci_integration/jenkins/client/model/jenkins_build.dart';
+import 'package:ci_integration/jenkins/client/model/jenkins_build_result.dart';
+import 'package:ci_integration/jenkins/client/model/jenkins_building_job.dart';
+import 'package:ci_integration/jenkins/client/model/jenkins_query_limits.dart';
+import 'package:metrics_core/metrics_core.dart';
+
+/// An adapter for [JenkinsClient] to fit [CiClient] contract.
+class JenkinsCiClientAdapter implements CiClient {
+  /// A Jenkins client instance used to perform API calls.
+  final JenkinsClient jenkinsClient;
+
+  /// Creates an instance of this adapter with the given [jenkinsClient].
+  ///
+  /// Throws [ArgumentError] if the given Jenkins client is `null`.
+  JenkinsCiClientAdapter(this.jenkinsClient) {
+    ArgumentError.checkNotNull(jenkinsClient, 'jenkinsClient');
+  }
+
+  @override
+  Future<List<BuildData>> fetchBuildsAfter(
+    String projectId,
+    BuildData build,
+  ) async {
+    ArgumentError.checkNotNull(build, 'build');
+
+    final buildingJob = await _fetchBuilds(
+      projectId,
+      limits: JenkinsQueryLimits.endBefore(0),
+    );
+    final lastBuild = buildingJob.lastBuild;
+    int numberOfBuilds = lastBuild.number - build.buildNumber;
+
+    if (numberOfBuilds <= 0) return [];
+
+    List<JenkinsBuild> builds;
+    while (builds == null) {
+      final buildingJob = await _fetchBuilds(
+        projectId,
+        limits: JenkinsQueryLimits.endAt(numberOfBuilds),
+      );
+
+      final _builds = buildingJob.builds;
+      if (_builds.isEmpty ||
+          _builds.first.number == buildingJob.firstBuild.number) {
+        builds = _builds;
+      } else {
+        final _earliestBuild = _builds.first;
+        final difference = _earliestBuild.number - build.buildNumber;
+
+        if (difference <= 0) {
+          builds = buildingJob.builds;
+        } else {
+          numberOfBuilds += difference;
+        }
+      }
+    }
+
+    return _processJenkinsBuilds(
+      builds,
+      buildingJob.name,
+      startFromBuildNumber: build.buildNumber,
+    );
+  }
+
+  @override
+  Future<List<BuildData>> fetchBuilds(String projectId) async {
+    final buildingJob = await _fetchBuilds(projectId);
+    return _processJenkinsBuilds(
+      buildingJob.builds,
+      buildingJob.name,
+    );
+  }
+
+  /// Fetches builds data for the project with given [projectId].
+  ///
+  /// The [limits] can be used to set a range-specifier for the request.
+  Future<JenkinsBuildingJob> _fetchBuilds(
+    String projectId, {
+    JenkinsQueryLimits limits = const JenkinsQueryLimits.empty(),
+  }) async {
+    final newBuildsFetchResult =
+        await jenkinsClient.fetchBuilds(projectId, limits: limits);
+    _throwIfInteractionUnsuccessful(newBuildsFetchResult);
+    return newBuildsFetchResult.result;
+  }
+
+  /// Checks the given [interactionResult] to be [InteractionResult.isSuccess].
+  ///
```

Comment:
```suggestion
```
nit: we probably don't need that initial as what comes next describes it well enough.

Reply:
I followed the Effective Dart documentation guide here: https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph. I think we should follow these rule.